spark-bigquery-dsv2/spark-bigquery-dsv2-common/src/main/java/com/google/cloud/spark/bigquery/v2/context/BigQueryDataSourceReaderContext.java

```diff
@@ -311,9 +324,28 @@ public void pruneColumns(StructType requiredSchema) {
   }
 
   public StatisticsContext estimateStatistics() {
-    return table.getDefinition().getType() == TableDefinition.Type.TABLE
-        ? new StandardTableStatisticsContext(table.getDefinition())
-        : UNKNOWN_STATISTICS;
+    createReadSession();
+    if (table.getDefinition().getType() == TableDefinition.Type.TABLE) {
```

Comment:
No need to create StandardTableDefinition for this. Statistics is a simple interface with two methods - you can create an anonymous implementation that returns the data from the readsession (or a simple class with two members

Reply:
Created anonymous implementation of StatisticsContext